File: shared/identity-generator/src/implementations/IdentityGeneratorNl.ts
import { IdentityGenerator } from "./base/IdentityGenerator";
import maleNames from '../dictionaries/nl/firstnames_male';
import femaleNames from '../dictionaries/nl/firstnames_female';
import lastNames from '../dictionaries/nl/lastnames';

/**
 * Identity generator for Dutch language using Dutch word dictionaries.
 */
export class IdentityGeneratorNl extends IdentityGenerator {
  /**
   * Get the male first names.
   */
  protected getFirstNamesMaleJson(): string[] {
    return maleNames;
  }

  /**
   * Get the female first names.
   */
  protected getFirstNamesFemaleJson(): string[] {
    return femaleNames;
  }

  /**
   * Get the last names.
   */
  protected getLastNamesJson(): string[] {
    return lastNames;
  }
}

File: apps/server/Generators/AliasVault.Generators.Identity/Implementations/IdentityGeneratorNl.cs
//-----------------------------------------------------------------------
// <copyright file="IdentityGeneratorNl.cs" company="lanedirt">
// Copyright (c) lanedirt. All rights reserved.
// Licensed under the AGPLv3 license. See LICENSE.md file in the project root for full license information.
// </copyright>
//-----------------------------------------------------------------------

namespace AliasVault.Generators.Identity.Implementations;

using AliasVault.Generators.Identity.Implementations.Base;

/// <summary>
/// Dutch identity generator which implements IIdentityGenerator and generates
/// random dutch identities.
/// </summary>
public class IdentityGeneratorNl : IdentityGenerator
{
    /// <inheritdoc cref="IdentityGenerator.FirstNamesListMale" />
    protected override string FirstNamesListMale => "AliasVault.Generators.Identity.Implementations.Dictionaries.nl.firstnames_male";

    /// <inheritdoc cref="IdentityGenerator.FirstNamesListFemale" />
    protected override string FirstNamesListFemale => "AliasVault.Generators.Identity.Implementations.Dictionaries.nl.firstnames_female";

    /// <inheritdoc cref="IdentityGenerator.LastNamesList" />
    protected override string LastNamesList => "AliasVault.Generators.Identity.Implementations.Dictionaries.nl.lastnames";
}
